src/components/expenses/CurrencySelect.tsx

```diff
@@ -0,0 +1,39 @@
+import { useTranslation } from 'react-i18next'
+import { FormControl, MenuItem } from '@mui/material'
+import { useField } from 'formik'
+
+import { Currency } from 'gql/currency'
+import FormTextField from 'components/common/form/FormTextField'
+
+export default function CurrencySelect() {
+  const name = 'currency'
```

Comment:
```suggestion
export default function CurrencySelect({ name = 'currency' }) {
```

If we move the name to the passed props with a default value we make this component really reusable

If we need it in a different form we can use it twice by passing different form `name` prop:

```tsx
<>
  <CurrencySelect name="fromCurrency" />
  <CurrencySelect name="toCurrency" />
</>
```

Reply:
Agree, and probably in the future should be moved to other place like `components/admin`. It'll be more _visible_ and we'll know that it's a component made to be shared across the admin forms.

I'll add the translation string to the passed props as well ```javascript
CurrencySelect({ name = 'currency', translation = 'expenses' })```. Just in case we decide to use different json file.

Reply:
Let's rename the translation prop from `translation` to `label` or something more explicit so it's clear what's the purpose.

It will be hard if we need to pass two `translation` props for different things

Reply:
@dpmarkov I do agree we can move this component to `components/currency` even now, no nee d to wait to a future moment

Reply:
Actually I was already confused what the `translation` actually means - it's the translation domain :dagger: not the actual label.

Please change it to `translationDomain` or `ttranslationNamespace`

Reply:
In that case should I remove `const { t } = useTranslation(translation)` and rely on just the label prop for that matter?

Reply:
I think we can keep it as the field label is getting translated automatically as `fields.${name}`

Normally for most of the fields we can have translations in `common.json` with key `fields`

No need to translate `currency` in every entity, as it would result in multiple translations for the same thing.

However if we pass the translation namespace/domain to the component we can override it which is a good thing for reusing it.

Does that make sense?

Reply:
@kachar maybe adding currency in the `common` translation domain might be a good idea?
If you agree it could be in common -> admin -> fields -> currency..

Reply:
I'll put 'fields' in the base of the json object not nested in admin. I didn't noticed at first that 'admin' is in 'nav'.